lib/src/platform_check/web.dart

```diff
@@ -13,11 +13,18 @@ class PlatformWeb extends Platform {
   static bool useBuiltInRng = false;
 
   PlatformWeb() {
+    useBuiltInRng = false;
     try {
       Random.secure();
       useBuiltInRng = true;
     } on UnsupportedError {
-      useBuiltInRng = false;
+      // Random.secure() normally throws this error if
+      // no cryptographically secure random number source is available.
+    } catch (e) {
+      // For Node.js with dart2js compiler, the following error is expected.
+      if (e.runtimeType.toString() == 'UnknownJsTypeError') {
+        // This error is internal to 'dart:_js_helper'.
+      }
```

Comment:
Suggestion: Rethrow on any unknown exception, this could be implemented with an else statement as well.

```suggestion
      // For Node.js with dart2js compiler, the UnknownJsTypeError error is expected.
      // This error is internal to 'dart:_js_helper' so we need to inspect the runtimeType.
      if (!(e.runtimeType.toString() == 'UnknownJsTypeError')) {
        rethrow
      }
```